libs/core-ui/src/lib/Cohort/ManualCohortManagement/CohortEditorUtils.ts

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { DatasetCohortColumns } from "../../DatasetCohortColumns";
+import { IDataset } from "../../Interfaces/IDataset";
+import { IExplanationModelMetadata } from "../../Interfaces/IExplanationContext";
+import { IFilter } from "../../Interfaces/IFilter";
+import { JointDataset } from "../../util/JointDataset";
+
+import { ICohortEditorProps } from "./CohortEditor";
+
+export function getFilters(
+  props: ICohortEditorProps,
+  dataset?: IDataset,
+  metadata?: IExplanationModelMetadata
+): IFilter[] {
+  if (props.filterList === undefined) {
+    return [];
+  }
+  if (props.isFromExplanation) {
+    return translateToNewFilters(props.filterList, metadata?.featureNames);
+  }
+  return translateToNewFilters(props.filterList, dataset?.feature_names);
+}
+
+export function translateToLegacyFilterColumn(
+  column: string,
+  featureNames: string[]
+): string {
+  if (column === DatasetCohortColumns.Index) {
```

Comment:
feel like this is better if using "switch case", but since this will be deprecated soon, either is fine to me!